### PR TITLE
[INJIMOB-787] Conditional restore of VCs

### PR DESCRIPTION
**Pull request by vharsh**
# Requirements

After partial VC restore, the VC restore should begin again automatically.

## Low level Implementation

1. If the ZIP file is corrupted, it should be deleted.
2. If the `ZIP / .injibackup` file is present, normal restore should automatically continue
3. If the restore ZIP/file is not present, normal flow should happen.
4. If the restore has happened partially and the app is killed, the restore should happen from that VC(implementation: partially downloaded VCs are removed & restore is started once again)

### CODE REVIEW

File: machines/backupRestore.ts [hunk 1]
```diff
@@ -236,11 +251,55 @@ export const backupRestoreMachine = model.createMachine(
...
+        if (bkpFile !== '') {
+          context.fileName = bkpFile;
+          // TODO: trim the extension later
+          return context.fileName;
+        }
+        // 3. Check for .zip files
+        const ZIPfile = await findMostRecentBackupFile();
+        if (ZIPfile !== '') {
+          // TODO: trim the extension later
+          context.fileName = getFileNameFromZIPfile(bkpFile);
```
Comment (vharsh) [inline machines/backupRestore.ts:275]: Set the filename with the .zip extension

File: shared/storage.ts [hunk 2]
```diff
@@ -106,12 +109,24 @@ class Storage {
...
+        this.unloadVCs(encryptionKey, parseInt(previousRestoreTS));
+      }
       // 1. opening the file
       const completeBackupData = JSON.parse(data);
       // 2. Load and store VC_records & MMKV things
+      previousRestoreTS = Date.now().toString();
       const dataFromDB = await Storage.loadVCs(
         completeBackupData,
         encryptionKey,
+        previousRestoreTS,
```
[the remark this answers is not in the document]
Comment (vharsh) [reply shared/storage.ts]: Assigned the value after updating the state of the `.prev` file. This file is always created on start of restore process and always killed right after the restore is done.

File: machines/backupRestore.ts [hunk 3]
```diff
@@ -45,6 +46,7 @@ export const BackupRestoreEvents = model.events;
...
...ERRFUABm5CsY+lZCSRtYsuR7niXEXkyT58n+Yp5TKcBNRqRIDSQZi0EzLp3QGUZzgmToZkoR4o7OHM8qAnY7yzNShFxcaCVkdgygAF5yLg6VUDxd78RWW4ubudYDcNo1SYxSkqQVdRFRBGmlVpMzAq4FiOB0riiMu2qneZ07WL0rVBroVhaE5kLxa5SULSNY2BcF5BhZF0WxTNL1zW9Q0fctMmrflyh9ptaLbc8bR7e1h22adp1Tg4rhKuYBI6e4J32H6sxmF1eyA71r25NGaXg1QTouuwdH4AI3KMxyAAytyw4KJUI68Oi2c4Fg6PhFKdNOJjeihYw9N0AvvEMFIjDs3Xk1GfV1jRbJVKgfAkNyYCseQRbXNwnC8KwAgAHI8Fz4FwxifOIHKvpqkuc7te4gK4Zd2GHa4mNDCdumuE9Rrq5TcZazresG0bZAm4IZtcPgzP4Opjuei74rmK4HseN7Jg+s4QvuDqtjjHMHShmTzlAyJSUQMUYBEIluRsnIZA3pN5YA3XFPA1TTcya3e4d2QkO9oV9s8-DnoPVolgnUSVjrg4RmKtLgZqt4oxewLp34X4YbKCQTfwK0glHMVc9lQAtFjKEP2Hz4RNEEk35nZW6OZS4LD0PRdCS1snnKwL8qwnCIJ-Qcu1F6agsBqGwfp1yOBBJdeBzUpRTm1HMPOG5Vb9wjoPPA0DNKI16EqJwCw3DEzqkSaywJwH1zfElOQEAZKkJ2ojQyQsdAbC9kuEmzUdKb2mH6IWsxjLkmwavGu4Y1Y7gblTD8jYTzNgvHmf80wHYwO4SsYW-CRgbAmHVYkxcFgbFsIAl2owwEEOegPJRcYxKeVpplPyXFOFOxmLYA6KwQR9C9vwouKFfQSPeLnEWAs5xMMcSwqm70losmkmALxno9TLmFuuAuCoTCuDzj6YE1hth53sr6fopN5GEMUfEuMFAabJMYmksqFdVSOFsuSWw+JLrDEWJqX47VDJlNiUQpx+5o5EF1vrQ2xtmm7T4c1awgIPDTi0EMRwPSLH9L9ArYZ9jw41NIqJZuo9DTjzmYjEYZdFg3UlgHCYy5LqGTVLYXQSx2r6TkVfUZtT9ywGwOgdAcAL46LIa8Uk9gEEeDGAM34WEnldG2LYewRIiRl0qd8w5bc4wRQvNgCgFzwXDGoQHPUlI5gDG8NjeY3R5h0ususykx8fBAA */
```
Comment (swatigoel) [inline machines/backupRestore.ts:49]: why this comment?

File: machines/backupRestore.ts [hunk 4]
```diff
@@ -262,6 +319,8 @@ export const backupRestoreMachine = model.createMachine(
     },
 
     guards: {
+      isBackupFile: (_, event) => event.data.endsWith('.injibackup'),
+      isZIPfile: (_, event) => event.data.endsWith('.zip'),
```
Comment (swatigoel) [inline machines/backupRestore.ts:323]: isZipfile not in use

File: machines/backupRestore.ts (hunk 3 above)
Comment (vharsh) [reply machines/backupRestore.ts:49]: this is auto-generated by Xstate

Comment (vharsh) [reply machines/backupRestore.ts:49]: In VS-code there's an extension which saves the layout of the boxes of the VC in here.

**Review by swatigoel — APPROVED** [review]

**Pull request merged**